test/x509_time_test.c

```diff
@@ -424,11 +424,121 @@ static int test_days(int n)
     return r;
 }
 
+static const struct {
+    ASN1_TIME asn1;
+    const char *readable;
+} x509_print_tests [] = {
+    {
```

Comment:
Could the contents of this array be produced via a macro:

```
#define M(time, type, date) { sizeof(time) - 1, type, (unsigned char *)time, 0 }, date

M("20170731222050Z", V_ASN1_GENERALIZEDTIME, "Jul 31 22:20:50 2017 GMT")
```

It would increase readability and is more concise.

With a better name instead of `M` and maybe a better laid out macro definition.